contrib/gdb_debugging_scripts/log_records.gdb

```diff
@@ -24,15 +24,8 @@ end
 # If page doesn't exist in cash, it will return 0
 #
 define logpb_get_page
-  set $hash = $arg0 % log_Pb.ht->size
-  set $hte = log_Pb.ht->table[$hash]
-  set $arg1 = 0
-  while $hte != 0
-    if *((LOG_PAGEID *) $hte->key) == $arg0
-      set $arg1 = &(((struct log_buffer *) ($hte->data))->logpage)
-      loop_break
-    end
-  end
+  set $idx = $arg0 % log_Pb.num_buffers
+  set $arg1 = log_Pb.buffers[$idx]->logpage

```

Comment:
@funkydvd 

I think to examine the pageid of the buffer matches with the given argument. 
Please look at the behavior of the old function.